miio/airconditioningcompanion.py

```diff
@@ -0,0 +1,99 @@
+from .device import Device
+import enum
+
+
+class OperationMode(enum.Enum):
+    Heating = 0
+    Cooling = 1
+    Auto = 2
+
+
+class FanSpeed(enum.Enum):
+    Low = 0
+    Medium = 1
+    High = 2
+    Auto = 3
+
+
+class AirConditioningCompanionStatus:
+    """Container for status reports of the Xiaomi AC Companion."""
+
+    def __init__(self, data):
+        # Device model: lumi.acpartner.v2
+        #
+        # Response of "get_model_and_state":
+        # ['010500978022222102', '010201190280222221', '2']
+        self.data = data
+
+    @property
+    def air_condition_power(self):
+        """Current power state of the air conditioner."""
+        return self.data[2]
+
+    @property
+    def air_condition_model(self):
+        """Model of the air conditioner."""
+        return str(self.data[0][0:2] + self.data[0][8:16])
+
+    @property
+    def power(self) -> str:
+        """Current power state."""
+        return 'on' if (self.data[1][2:3] == 1) else 'off'
+
+    @property
+    def is_on(self) -> bool:
+        """True if the device is turned on."""
+        return self.power == 'on'
+
+    @property
+    def temperature(self) -> int:
+        """Current temperature."""
+        return int(self.data[1][6:8], 16)
+
+    @property
+    def swing_mode(self) -> bool:
+        """True if swing mode is enabled."""
+        return self.data[1][5:6] == '0'
+
+    @property
+    def fan_speed(self):
+        """Current fan speed."""
+        return OperationMode(self.data[1][4:5])
+
+    @property
+    def mode(self):
+        """Current operation mode."""
+        return OperationMode(self.data[1][3:4])
+
+
+class AirConditioningCompanion(Device):
+    """Main class representing Xiaomi Air Conditioning Companion."""
+
+    def status(self) -> AirConditioningCompanionStatus:
+        """Return device status."""
+        status = self.send("get_model_and_state", [])
+        return AirConditioningCompanionStatus(status)
+
+    def learn(self):
+        """Learn an infrared command."""
+        return self.send("start_ir_learn", [30])
```

Comment:
The magic 30 should be extracted into a constant and/or documented some other way.